File: src/Tests/Behavioral/GenericTypeDecl/GenericTypeDecl.go
package main

import "fmt"

// Pair is a generic type with two type parameters
type Pair[T, U any] struct {
	First  T
	Second U
}

func main() {
	// Instantiate the generic type with concrete types
	p := Pair[string, int]{
		First:  "answer",
		Second: 42,
	}

	fmt.Printf("Pair: %v, %v\n", p.First, p.Second)
}

File: src/Tests/Behavioral/GenericTypeDecl/GenericTypeDecl.cs
namespace go;

using fmt = fmt_package;

partial class main_package {

[GoType] partial struct Pair<T, U>
    where T : new()
    where U : new()
{
    public T First;
    public U Second;
}

private static void Main() {
    var p = new Pair<@string, nint>(
        First: "answer"u8,
        Second: 42
    );
    fmt.Printf("Pair: %v, %v\n"u8, p.First, p.Second);
}

} // end main_package
